tests/Backend/ExternalBuckets/BigqueryRegisterBucketTest.php

```diff
@@ -0,0 +1,525 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\Test\Backend\ExternalBuckets;
+
+use Google\Cloud\BigQuery\AnalyticsHub\V1\AnalyticsHubServiceClient;
+use Google\Cloud\BigQuery\AnalyticsHub\V1\DataExchange;
+use Google\Cloud\BigQuery\AnalyticsHub\V1\Listing;
+use Google\Cloud\BigQuery\AnalyticsHub\V1\Listing\BigQueryDatasetSource;
+use Google\Cloud\BigQuery\BigQueryClient;
+use Google\Cloud\Iam\V1\Binding;
+use GuzzleHttp\Client;
+use Keboola\StorageApi\ClientException;
+use Keboola\StorageApi\Workspaces;
+use Keboola\Test\Backend\Workspaces\Backend\BigQueryClientHandler;
+use Keboola\Test\Backend\Workspaces\Backend\WorkspaceBackendFactory;
+use Keboola\Test\Utils\EventsQueryBuilder;
+
+class BigqueryRegisterBucketTest extends BaseExternalBuckets
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+    }
+
+    public function testRegisterBucket(): void
+    {
+        $this->initEvents($this->_client);
+        $token = $this->_client->verifyToken();
+
+        if (!in_array('input-mapping-read-only-storage', $token['owner']['features'])) {
+            $this->markTestSkipped(sprintf('Read only mapping is not enabled for project "%s"', $token['owner']['id']));
+        }
+        if (!in_array('external-buckets', $token['owner']['features'])) {
+            $this->markTestSkipped(sprintf('External buckets are not enabled for project "%s"', $token['owner']['id']));
+        }
+        $this->allowTestForBackendsOnly([self::BACKEND_BIGQUERY], 'Backend has to support external buckets');
+        $this->expectNotToPerformAssertions();
+    }
+
+    public function testInvalidListingToRegister(): void
+    {
+        $this->dropBucketIfExists($this->_client, 'in.test-bucket-registration', true);
+
+        try {
+            $this->_client->registerBucket(
+                'test-bucket-registration',
+                ['invalid-google-url'],
+                'in',
+                'will fail',
+                'bigquery',
+                'test-bucket-will-fail'
+            );
+            $this->fail('should fail');
+        } catch (ClientException $e) {
+            $this->assertSame('storage.buckets.validation', $e->getStringCode());
+            $this->assertStringContainsString(
+                'Not valid listing url.',
```

Comment:
Nešlo by tady dát komplet tu error message? Aby bylo vidět, že dává smysl celkově?

Reply:
Ale to je cela ta err msg

Reply:
Aha, sorry, tak bych byl pro to upravit buď `"IX" is not a valid listing URL` nebo aspoň `Invalid listing URL`. Kdybysme to teď pustili ven, tak pak budem řešit, že změna message je bc break :/

Reply:
Ok navrhujem teda `Invalid listing URL` pride mi ze to tam vypisovat to co som dal na inpute mi pride zbytocne ked viem co som tam posielal.

Reply:
Jo, klidně.

Reply:
Hotovo @tomasfejfar